pkg/sql/crdb_internal.go

```diff
@@ -1533,3 +1537,111 @@ CREATE TABLE crdb_internal.zones (
 		return nil
 	},
 }
+
+// crdbInternalGossipNodes exposes local information about the cluster nodes.
+var crdbInternalGossipNodes = virtualSchemaTable{
+	schema: `
+CREATE TABLE crdb_internal.gossip_nodes (
+  node_id  INT NOT NULL,
+  network  STRING NOT NULL,
+  address  STRING NOT NULL,
+  attrs    STRING NOT NULL,
+  locality STRING NOT NULL,
+  version  STRING NOT NULL
+)
+	`,
+	populate: func(ctx context.Context, p *planner, prefix string, addRow func(...tree.Datum) error) error {
+		if err := p.RequireSuperUser("read crdb_internal.gossip_nodes"); err != nil {
+			return err
+		}
+
+		g := p.session.execCfg.Gossip
+		var descriptors []roachpb.NodeDescriptor
+		if err := g.IterateInfos(gossip.KeyNodeIDPrefix, func(key string, i gossip.Info) error {
+			bytes, err := i.Value.GetBytes()
+			if err != nil {
+				return errors.Wrapf(err, "failed to extract bytes for key %q", key)
+			}
+
+			var d roachpb.NodeDescriptor
+			if err := protoutil.Unmarshal(bytes, &d); err != nil {
+				return errors.Wrapf(err, "failed to parse value for key %q", key)
+			}
+			descriptors = append(descriptors, d)
+			return nil
+		}); err != nil {
+			return err
+		}
+
+		sort.Slice(descriptors, func(i, j int) bool {
+			return descriptors[i].NodeID < descriptors[j].NodeID
+		})
+
+		for _, d := range descriptors {
+			if err := addRow(
+				tree.NewDInt(tree.DInt(d.NodeID)),
+				tree.NewDString(d.Address.NetworkField),
+				tree.NewDString(d.Address.AddressField),
+				tree.NewDString(d.Attrs.String()),
```

Comment:
Actually, as a toy test for https://github.com/cockroachdb/cockroach/issues/20713#issuecomment-353689807, try turning `attrs` into a JSON field.

I think you'll need `tree.MakeDJSON(s)` where `s` is an `[]interface{}` with each entry being backed by a `string` (`s := "myattr", s = append(s, interface{}{s})`).